modules/govuk_jenkins/manifests/safe_restart.pp

```diff
@@ -0,0 +1,38 @@
+# == Class: Govuk_jenkins::Safe_restart
+#
+# This class will issue a "safe restart" of Jenkins, which means it will only
+# restart once all jobs have finished running.
+#
+# Note: this requires having a user enabled that has the appropriate public SSH
+# key assigned to it. This should be added manually in advance otherwise the
+# command will fail.
+#
+# === Parameters:
+#
+# [*jenkins_home*]
+#   The home directory of the Jenkins install, and where to put the Jenkins CLI
+#   jar.
+#
+# [*jenkins_user*]
+#   Which user to run the jar as.
+#
+class govuk_jenkins::safe_restart (
+  $jenkins_home = $::govuk_jenkins::jenkins_homedir,
+  $jenkins_user = $::govuk_jenkins::jenkins_user,
+) {
+  require ::govuk_jenkins
+
+  exec { 'Jenkins CLI tool':
```

Comment:
I think that would be nice, but it doesn't give the ability to specify the user/group that the file gets set as. I'm wondering if it needs to be run as Jenkins user at all, actually.